### PR TITLE
DFCxx -> SystemVerilog pipeline parametrization

### PR DESCRIPTION
**Pull request by Muxianesty**
Mentioned in #23 .

DFCxx now supports different output paths for different artifacts.
Now an output path for SystemVerilog has to be set explicitly (option `--out <PATH>`)

### CODE REVIEW

**Review by ssmolov — COMMENTED** [review]
When applying this PR, I receive a segmentation fault upon the following usage of the tool:
```console
build/src/umain hls --config examples/polynomial2/polynomial2.json -l
```

**Comment by Muxianesty** [comment]
> When applying this PR, I receive a segmentation fault upon the following usage of the tool:
> 
> ```
> build/src/umain hls --config examples/polynomial2/polynomial2.json -l
> ```

I mentioned this in the PR's description: Now an output path for SystemVerilog has to be set explicitly (option `--out <PATH>`).
I didn't include this in `README.md`, because this PR is intermediate and `README.md` will change in #22 .

File: src/model/dfcxx/lib/dfcxx/converter.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ bool DFCIRConverter::convertAndPrint(mlir::ModuleOp module,
   }
   pm.addPass(circt::createLowerFIRRTLToHWPass());
   pm.addPass(circt::createLowerSeqToSVPass());
-  pm.addPass(circt::createExportVerilogPass(out));
+  pm.addPass(circt::createExportVerilogPass(*(outputStreams[SV_OUT_ID])));
```
Comment (Muxianesty) [inline src/model/dfcxx/lib/dfcxx/converter.cpp:40]: In the future every single access to `outputStreams` will be preceded by an `if`-statement.

File: src/model/dfcxx/lib/dfcxx/kernel.cpp [hunk 2]
```diff
@@ -40,23 +40,31 @@ DFType Kernel::dfBool() {
...
                      const Scheduler &sched) {
-  if (filePath.empty()) { return compile(config, sched); }
   DFCIRBuilder builder(config);
   auto compiled = builder.buildModule(this);
-  std::error_code ec;
-  llvm::raw_fd_ostream out(filePath, ec);
-  bool result = DFCIRConverter(config).convertAndPrint(compiled, out, sched);
-  out.close();
+  size_t count = outputPaths.size();
+  std::vector<llvm::raw_fd_ostream *> outputStreams(count);
```
Comment (Muxianesty) [inline src/model/dfcxx/lib/dfcxx/kernel.cpp:50]: `llvm::raw_fd_ostream` isn't copiable, thus pointers are used.

File: src/model/dfcxx/include/dfcxx/kernel.h [hunk 3]
```diff
@@ -57,9 +58,8 @@ class Kernel {
 
   virtual std::string_view getName() = 0;
 
-  bool compile(const DFLatencyConfig &config, const Scheduler &sched);
```
Comment (Muxianesty) [inline src/model/dfcxx/include/dfcxx/kernel.h:60]: Old functionality to dump directly to `stdout` was removed.

**Review by ssmolov — APPROVED** [review]

**Pull request merged**